code/modules/mob/dead/new_player/preferences_setup.dm

```diff
@@ -46,6 +46,51 @@
 	if(randomise[RANDOM_NAME])
 		real_name = pref_species.random_name(gender,1)
 
+///Setup a hardcore random character and calculate their hardcore random score
+/datum/preferences/proc/hardcore_random_setup(mob/living/carbon/human/character, antagonist)
+	random_character(TRUE, antagonist)
+	select_hardcore_quirks()
+	hardcore_survival_score = hardcore_survival_score ** 1.2 //30 points would be about 60 score
+
+///Go through all quirks that can be used in hardcore mode and select some based on a random budget.
+/datum/preferences/proc/select_hardcore_quirks()
+
+	var/quirk_budget = rand(8, 35)
+
+
+	all_quirks = list() //empty it out
+
+	var/list/available_hardcore_quirks = SSquirks.hardcore_quirks.Copy()
+
+	while(quirk_budget > 0)
+		for(var/i in available_hardcore_quirks) //Remove from available quirks if its too expensive.
+			var/datum/quirk/T = i
+			if(available_hardcore_quirks[T] > quirk_budget)
+				available_hardcore_quirks -= T
+
+		var/datum/quirk/picked_quirk = pick(available_hardcore_quirks)
+
+		for(var/bl in SSquirks.quirk_blacklist) //Check if the quirk is blacklisted. V is a list
+			var/list/blacklist = bl
+			for(var/iterator_quirk in all_quirks)
+				if((picked_quirk in blacklist) && (iterator_quirk in blacklist) && !(iterator_quirk == picked_quirk)) //two quirks have lined up in the list of the list of quirks that conflict with each other, so return (see quirks.dm for more details)

```

Comment:
`(picked_quirk in blacklist)` can be checked outside of the for loop, continuing early.

Reply:
cool good idea I'll apply this to OG blacklist code too